code/obj/item/nuclear_operative/surplus_ops_items.dm

```diff
@@ -0,0 +1,194 @@
+			new helm(src.loc)
+			..()
+
+
+
+//Items for surplusops that don't belogn anywhere else
+/obj/item/reagent_containers/glass/beaker/large/surplusmedical
+	name = "Doctor Schmidt's Super Mega Restoration Jungle Juice"
+	desc = "A beaker containing a supposed panacea. It smells weird and the glass feels sticky."
+	initial_reagents = list("ephedrine"=30, "saline"= 30, "synaptizine" = 30, "omnizine" = 9)
+
+/obj/item/experimental/melee/spear/plaswood
+	New()
+		..()
+		setHeadMaterial(getMaterial("plasmaglass"))
+		setShaftMaterial(getMaterial("wood"))
+		buildOverlays()
+
+/obj/machinery/surplusopauth
+	name = "Deployment Authorization Computer"
+	desc = "A voting computer that allows three users to activate a certain portal frame."
+	icon_state = "drawbr"
+	icon = 'icons/obj/computer.dmi'
+	density = 0
+
+	var/obj/submachine/surplusopdeployer/ourportal = null
+	var/auth_need = 3
+	var/list/authorized
+	var/list/authorized_registered = null
+
+	var/authed = 0
+
+
+	New()
+		for_by_tcl(D, /obj/submachine/surplusopdeployer)
+			src.ourportal = D //connect to portal
+		desc = "A voting computer that allows three users to activate a certain portal frame. [auth_need] votes are left."
+		..()
+
+
+
+//account for both empty and full hand authorizations
+	attackby(var/obj/item/W, var/mob/user)
+		authaction(user)
+		..()
+	attack_hand(mob/user)
+		authaction(user)
+		..()
+// what goes down when you try to auth- stolen from armory code
+	proc/authaction(var/mob/user)
+
+		if (!user)
+			return
+		if (!src.authorized)
+			src.authorized = list()
+			src.authorized_registered = list()
+
+		var/choice = tgui_alert(user, "Would you like to authorize group deployment? [src.auth_need - length(src.authorized)] authorization\s are still needed.", "Armory Auth", list("Authorize"))
+		if(BOUNDS_DIST(user, src) > 0 || src.authed)
+			return
+		src.add_fingerprint(user)
+		if (!choice)
+			return
+		switch(choice)
+			if("Authorize")
+				if (user in src.authorized)
+					boutput(user, "You have already authorized! [src.auth_need - src.authorized.len] authorizations from others are still needed.")
+					return
+				//we only want fingerprints to be able to count, since they don't have any access
+				if (ishuman(user))
+					var/mob/living/carbon/human/H = user
+					if (H.bioHolder.Uid in src.authorized)
+						boutput(user, "You have already authorized - fingerprints on file! [src.auth_need - src.authorized.len] authorizations from others are still needed.")
+						return
+					src.authorized += H.bioHolder.Uid
+				else
+					src.authorized += user
+
+
+				if (src.authorized.len < auth_need)
+					print_auth_needed(user)
+				else
+					authorize()
+
+	proc/authorize()
+
+		if(src.authed)
+			return
+
+		src.visible_message("<span class='alert'><b>Deployment has been authorized! Group up and deploy!</b></span>")
+		for(var/obj/submachine/surplusopdeployer/D in range(get_turf(src), 16)) //this is insanely stupid but for_by_tcl was misbehaving
```

Comment:
`for_by_tcl` requires `START_TRACKING` and `STOP_TRACKING` macros in `New` and `disposing`, probably why it was misbehaving?